service/collector.go

```diff
@@ -99,14 +100,26 @@ func New(set CollectorSettings) (*Collector, error) {
 	}
 
 	col := &Collector{
-		info:         set.BuildInfo,
-		factories:    set.Factories,
-		stateChannel: make(chan State, Closed+1),
+		info:           set.BuildInfo,
+		factories:      set.Factories,
+		stateChannel:   make(chan State, Closed+1),
+		parserProvider: set.ParserProvider,
+		configLoader:   set.ConfigLoader,
```

Comment:
Are you keeping `set.ConfigLoader` to avoid a breaking change on the struct?

Reply:
Done.